nixos/modules/services/networking/amuled.nix

```diff
@@ -66,15 +68,37 @@ in
       wantedBy = [ "multi-user.target" ];
       after = [ "network.target" ];
 
-      preStart = ''
-        mkdir -p ${cfg.dataDir}
-        chown ${user} ${cfg.dataDir}
-      '';
-
-      script = ''
-        ${pkgs.su}/bin/su -s ${pkgs.runtimeShell} ${user} \
-            -c 'HOME="${cfg.dataDir}" ${pkgs.amule-daemon}/bin/amuled'
-      '';
+      serviceConfig = {
+        Environment = [ "HOME=${cfg.dataDir}" ];
+        Type = "forking";
+        User = "${cfg.user}";
+        WorkingDirectory = cfg.dataDir;
```

Comment:
There is nothing making sure this path exists. Note that systemd will consider this a fatal error and it will refuse to start the service.

Perhaps use `systemd.tmpfiles`?

Reply:
Added in https://github.com/NixOS/nixpkgs/pull/354517/commits/6b565fd1c5f732f6a2c3ee7537dafbd9e16c160b, thanks.